Cogs.Publishers/UmlSchemaPublisher.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) 2017 Colectica. All rights reserved
+// See the LICENSE file in the project root for more information.
+using Cogs.Model;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml.Linq;
+using System.Xml;
+using System.Diagnostics;
+
+namespace Cogs.Publishers
+{
+    /// <summary>
+    /// Generate an uml schema using the Garden of Eden approach, all elements and type definitions are declared globally
+    /// </summary>
+    public class UmlSchemaPublisher
+    {
+        // path to outer file to read into 
+        public string CogsLocation { get; set; }
+        // path to write output in
+        public string TargetDirectory { get; set; }
+        // boolean to determine whether to replace existing or not
+        public bool Overwrite { get; set; }
+
+        // list of all IDs created. Used to ensure no duplicates
+        private List<string> IdList = new List<string>();
+
+        public void Publish(CogsModel model)
+        {
+            if (CogsLocation == null)
+            {
+                throw new InvalidOperationException("Cogs location must be specified");
+            }
+            if (TargetDirectory == null)
+            {
+                throw new InvalidOperationException("Target directory must be specified");
+            }
+            if (Overwrite && Directory.Exists(TargetDirectory))
+            {
+                Directory.Delete(TargetDirectory, true);
+            }
+            // what about Overwrite being false and Directory.Exists(TargetDirectory))?
```

Comment:
We can return an error and exit. We still have to define return codes for the main program and that could be one of them. Leave the comment until we implement that, perhaps using a //TODO ...